Source/UINavigation/Private/UINavWidget.cpp

```diff
@@ -1263,20 +1279,19 @@ void UUINavWidget::ReplaceButtonInNavigationGrid(UUINavButton * ButtonToReplace,
 
 void UUINavWidget::UpdateCurrentButton(UUINavButton * NewCurrentButton)
 {
-	ButtonIndex = NewCurrentButton->ButtonIndex;
 	if (IsSelectorValid())
 	{
-		if (MoveCurve != nullptr) BeginSelectorMovement(NewCurrentButton->ButtonIndex);
-		else UpdateSelectorLocation(NewCurrentButton->ButtonIndex);
+		UpdateSelectorWaitForTick = 0;
+		UpdateSelectorPrevButtonIndex = ButtonIndex;
+		UpdateSelectorNextButtonIndex = NewCurrentButton->ButtonIndex;
+		bUpdateSelector = true;
 	}
 
+	ButtonIndex = NewCurrentButton->ButtonIndex;
+	
 	for (UScrollBox* ScrollBox : ScrollBoxes)
 	{
-		if (NewCurrentButton->IsChildOf(ScrollBox))
```

Comment:
Why did you remove this?

Reply:
Reading Unreal engine source code, SSCrollBox.cpp function bool SScrollBox::InternalScrollDescendantIntoView(), called from ScrollDescendantIntoView(), called from ScrollBox.cpp ScrollWidgetIntoView() is already checking if the widget is a descendant of the scrollbox and printing a warning message if it isn't. So The check is performed twice for each button descendant of the scrollbox.

Reply:
I left it as it was to avoid the annoying message.